charts/agent/templates/daemonset.yaml

```diff
@@ -29,8 +32,8 @@ spec:
       {{- end }}
     spec:
       serviceAccountName: {{ template "agent.serviceAccountName" .}}
-{{- if .Values.priorityClassName }}
-      priorityClassName: "{{ .Values.priorityClassName }}"
+{{- if or .Values.priorityClassName .Values.gke.autopilot }}
```

Comment:
use `{{- include "agent.gke.autopilot" . }}` to check for autopilot using the helper.